src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useMemo, useCallback } from 'react';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
+import { Person } from '../types/Person';
+
+type Props = {
+  people: Person[];
+  selectedPerson?: Person | null;
+  onSelect?: (person: Person) => void;
+};
+
+export const AutoComplete: React.FC<Props> = ({
+  people,
+  selectedPerson,
+  onSelect = () => { },
+}) => {
+  const [visibleList, setVisibleList] = useState(false);
+  const [query, setQuery] = useState(selectedPerson?.name || '');
+  const [applieQuery, setApplieQuery] = useState('');
+
+  const applyQuery = useCallback(
```

Comment:
let's implement this part:

![image](https://github.com/mate-academy/react_autocomplete/assets/86923089/a0504212-7539-4cdf-8809-f220a994f969)